### DISCUSSION TITLE
Passing props to another page using Router

### DISCUSSION THREAD

**Discussion by jessecheu** [vercel/next.js#17008] [Help] [answered]
Hello, I recently switched from ReactJS to NextJS App and I am trying to figure out how to pass Props to another page. 
I tried this on NextJS but it is passing it as a query on the web URL but I need it as a props so It cant be edited and doing state : {} will not pass also. I was thinking about using Redux to pass props between pages but I am hoping that there's a easier way to pass props using NextJs's Router. 

```
 Router.push({
                pathname: '/checkout-summary',
                query: {
                    rate: this.state.rate,
                    state: {
                        data
                    }
                }
```

another try:

```
            Router.push({
                pathname: '/checkout-summary',
                query: {
                    rate: this.state.rate,
                },
                state: {
                    data
                }
            })
```

on ReactJS I was doing this to pass data between pages.

```
this.props.history.push({
                pathname:'/checkout-summary',
                state: {
                    data
                }
```

Any help would be greatly appreciated.

**Comment by jamesmosier** [comment, marked as answer]
Hi @jessecdob. There is no way to pass data between pages with Next's router. You would have to either use query strings as you mentioned, sessionStorage, or a state management option like Redux or React's Context API. You could then put that in your `pages/_app.js` file which would wrap all pages so then you'd have a centralized state management solution.

**Reply by landed1** [reply]
I wanted to pass props from the main wrapper app to my component 'body' is that possible? Is it because we use functions instead of React Component?

**Reply by landed1** [reply]
I got the context working. Though the implementation seems to have duplication it works with some effort and probably can solve most cases without needing REDUX.

**Comment by jmhungdev** [comment]
For people that still want to pass data between pages using router, here's one way to do it, it works for me.
https://www.freecodecamp.org/news/routing-in-nextjs-beginners-guide/

**Reply by jason-shen** [reply]
i think what he is after is not this query string method, as this looks bad, that you have all your data in a url

**Reply by Subi** [reply]
Yeah, this is what I am trying to avoid as well.

**Comment by Lambtsa** [comment]
Another solution would be to use context. I wanted to handle passing requested url to login page so that i could redirect user to correct place after successful login.

**Reply by wixbaruch** [reply]
React-Context it is relevant only for clientSide renderring, but if you want to access to the context from SSR (for example from getServerSideProps function) you cannot do it

**Reply by popisdead** [reply]
Absolutely not true.
You can pass data from a page to another through a form action with a POST method, and parse the body through
`import { parseBody } from "next/dist/server/api-utils/node";`
then in:
```
getServerSideProps(context) {
    if (context.req.method === "POST") {
        const body = await parseBody(context.req);
```
Here's how you can do that in SSR - this has been done for decades - and it is the basic of client / server

**Reply by wixbaruch** [reply]
I meant to [React.Context ](https://reactjs.org/docs/context.html) ,and when you routing between pages from SSR you cannot to access to same context that you created in client-side (only if you send the context data in the body, but this is influnce on the performance, this is depends on your needs)
